pkg/process/checks/net.go

```diff
@@ -126,12 +132,85 @@ func (c *ConnectionsCheck) Realtime() bool { return false }
 // ShouldSaveLastRun indicates if the output from the last run should be saved for use in flares
 func (c *ConnectionsCheck) ShouldSaveLastRun() bool { return false }
 
-// Run runs the ConnectionsCheck to collect the active network connections
-// and any closed network connections since the last Run.
-// For each connection we'll return a `model.Connection`
-// that will be bundled up into a `CollectorConnections`.
-// See agent.proto for the schema of the message and models.
-func (c *ConnectionsCheck) Run(nextGroupID func() int32, _ *RunOptions) (RunResult, error) {
+func (c *ConnectionsCheck) runGRPC(nextGroupID int32) (RunResult, error) {
+	// Create a context with a timeout of 10 seconds
+	timedContext, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	opts := []grpc.DialOption{
+		grpc.WithContextDialer(func(_ context.Context, _ string) (nnet.Conn, error) {
+			return nnet.Dial(api.NetType, c.syscfg.GRPCSocketFilePath)
+		}),
+		grpc.WithInsecure(),
+	}
+
+	conn, err := grpc.DialContext(timedContext, "unused", opts...)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	client := connectionserver.NewSystemProbeClient(conn)
+
+	response, err := client.GetConnections(timedContext, &connectionserver.GetConnectionsRequest{ClientID: c.tracerClientID}, grpc.MaxCallRecvMsgSize(api.MaxGRPCSererMessage), grpc.MaxCallSendMsgSize(api.MaxGRPCSererMessage))
+	if err != nil {
+		return nil, err
+	}
+
+	// We only need to fetch once for a batch, and this boolean indicates that.
+	processFetchSucceeded := true
+	if err := c.processData.Fetch(); err != nil {
+		processFetchSucceeded = false
+		log.Warnf("error collecting processes for filter and extraction: %s", err)
+	}
+
+	unmarshaler := netEncoding.GetUnmarshaler(netEncoding.ContentTypeProtobuf)
+	var batchMessages []model.MessageBody
+
+	for {
+		start := time.Now()
+		res, err := response.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		batch, err := unmarshaler.Unmarshal(res.Data)
```

Comment:
Once you have `batch`, isn't the rest of the loop the same as `runHTTP`? Can we reuse the code there?

Reply:
Agreed, I have created 2 functions that includes the shard code, one which is handling the connections and one for handling the telemetry, then I have used the share code for each - HTTP and gRPC